dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04330WorkflowTaskAddLanguageIdColumn.java

```diff
@@ -0,0 +1,275 @@
+            DbType.MYSQL,        "UPDATE workflow_task SET language_id = ?",
+            DbType.ORACLE,       "UPDATE workflow_task SET language_id = ?",
+            DbType.MSSQL,        "UPDATE workflow_task SET language_id = ?"
+    );
+
+    private static final Map<DbType, String> addLanguageIdIndexSQLMap       = map(
+            DbType.POSTGRESQL,   "create index idx_workflow_6 on workflow_task (language_id)",
+            DbType.MYSQL,        "create index idx_workflow_6 on workflow_task (language_id)",
+            DbType.ORACLE,       "create index idx_workflow_6 on workflow_task (language_id)",
+            DbType.MSSQL,        "create index idx_workflow_6 on workflow_task (language_id)"
+    );
+
+    private static final Map<DbType, String> addLanguageIdForeignKeySQLMap       = map(
+            DbType.POSTGRESQL,   "alter table workflow_task add constraint FK_workflow_task_language foreign key (language_id) references language(id)",
+            DbType.MYSQL,        "ALTER TABLE workflow_task ADD CONSTRAINT FK_workflow_task_language FOREIGN KEY (language_id) REFERENCES language(id)",
+            DbType.ORACLE,       "alter table workflow_task add constraint FK_workflow_task_language foreign key (language_id) references language(id)",
+            DbType.MSSQL,        "ALTER TABLE workflow_task ADD CONSTRAINT FK_workflow_task_language FOREIGN KEY (language_id) REFERENCES language(id)"
+    );
+
+    private static final long INVALID_LANGUAGE_ID = -1;
+    @Override
+    public boolean forceRun() {
+        return Boolean.TRUE;
+    }
+
+    @Override
+    public void executeUpgrade() throws DotDataException {
+
+
+        final boolean created = this.addLanguageIdColumn();
+
+        if (created) {
+
+            if (DbConnectionFactory.isMsSql() && DbConnectionFactory.getAutoCommit()) {
+                DbConnectionFactory.setAutoCommit(false); // set a transactional for data
+            }
+
+            this.updateAllWorkflowTaskToDefaultLanguage();
+
+            // if mssql is in a transaction for the data, commit, close and start a new one.
+            if (DbConnectionFactory.isMsSql() && !DbConnectionFactory.getAutoCommit()) {
+                this.closeCommitAndStartTransaction();
+                DbConnectionFactory.setAutoCommit(true);
+            }
+
+            this.addWorkflowTaskLanguageIdIndex();
+            this.addWorkflowTaskLanguageIdFK();
+        }
+    }
+
+    private boolean addLanguageIdColumn() throws DotDataException {
+
+        boolean needToCreate = false;
+        Logger.info(this, "Adding new 'language_id' column to 'workflow_task' table.");
+
+        try {
+
+            if (DbConnectionFactory.isMsSql() && !DbConnectionFactory.getAutoCommit()) {
+                DbConnectionFactory.setAutoCommit(true);
+            }
+
+            new DotConnect().setSQL(this.getSelectLanguageIdColumnSQL()).loadObjectResults();
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)